test/integration_test/applicationbackup_test.go

```diff
@@ -55,36 +78,56 @@ func triggerBackupRestoreTest(
 	}
 
 	// Backup application
-	err = schedulerDriver.AddTasks(ctxs[0],
-		scheduler.ScheduleOptions{AppKeys: appBackupKey})
+	if !backupAllAppsExpected && backupSuccessExpected {
```

Comment:
This is a little confusing, can you add comments here on what this is trying to do and what all the context variables are supposed to track.

Reply:
I would suggest moving these to just before the corresponsing WaitForRunning() on lines 90,92 and 118 to make it simpler

Reply:
Fixed.